nixos/modules/programs/java.nix

```
@@ -55,9 +52,10 @@ in {
         recognitionType = "extension";
         magicOrExtension = "class";
         interpreter = pkgs.writeShellScript "java-class-wrapper" ''
-          file=''${1%%.class}
-          file=''${file/.\//}
-          java $file
+          test -e ${cfg.package}/nix-support/setup-hook && source ${cfg.package}/nix-support/setup-hook
+          classpath=$(dirname "$1")
+          class=$(basename "''${1%%.class}")
+          $JAVA_PATH -classpath "$classpath" "$class"
```

```suggestion
          $JAVA_PATH -classpath "$classpath" "$class" "''${@:2}"
```

One more small nit, to make sure command args get passed through (I apologize if I'm being too nitpicky)

---

And done